File: ea/src/evolution_engine/evolution/EvolutionParameters.java
package evolution_engine.evolution;

import evolution_engine.EvolutionLogger;
import evolution_engine.fitness.FitnessCalculator;
import evolution_engine.mutators.Crossover;
import evolution_engine.mutators.Mutator;
import evolution_engine.selection.SelectionStrategy;

import java.util.List;

public class EvolutionParameters<I extends Individual> {
    private final FitnessCalculator fitnessCalculator;
    private final SelectionStrategy selectionStrategy;
    private final Crossover<I> crossover;
    private final List<? extends Mutator> mutators;
    private final PopulationInitializer<I> populationInitializer;
    private int populationSize;


    private EvolutionLogger logger = null;
    private int generations;

    public EvolutionParameters(FitnessCalculator fitnessCalculator, SelectionStrategy selectionStrategy, Crossover<I> crossover, List<? extends Mutator> mutators, PopulationInitializer<I> populationInitializer, int generations, int populationSize) {
        this.fitnessCalculator = fitnessCalculator;
        this.selectionStrategy = selectionStrategy;
        this.crossover = crossover;
        this.mutators = mutators;
        this.populationInitializer = populationInitializer;
        this.generations = generations;
        this.populationSize = populationSize;
    }

    public FitnessCalculator getFitnessCalculator() {
        return fitnessCalculator;
    }

    public SelectionStrategy getSelectionStrategy() {
        return selectionStrategy;
    }

    public Crossover<I> getCrossover() {
        return crossover;
    }

    public List<? extends Mutator> getMutators() {
        return mutators;
    }

    public PopulationInitializer<I> getPopulationInitializer() {
        return populationInitializer;
    }

    public boolean isLoggingEnable() {
        return logger != null;
    }

    public EvolutionLogger getLogger() {

        return logger;
    }

    public void setLogger(EvolutionLogger logger) {
        this.logger = logger;
    }

    public int getPopulationSize() {
        return populationSize;
    }

    public void setPopulationSize(int populationSize) {
        this.populationSize = populationSize;
    }

    public int getGenerations() {
        return generations;
    }

    public void setGenerations(int generations) {
        this.generations = generations;
    }
}

File: src/evolution_engine/evolution/EvolutionParameters.scala
package evolution_engine.evolution

import evolution_engine.EvolutionLogger
import evolution_engine.fitness.FitnessCalculator
import evolution_engine.mutators.{Crossover, Mutator}
import evolution_engine.selection.SelectionStrategy

class EvolutionParameters[I <: Individual] {
  private final var fitnessCalculator: FitnessCalculator[I] = null
  private final var selectionStrategy: SelectionStrategy[I] = null
  private final var crossover: Crossover[I] = null
  private final var mutators: List[_ <: Mutator[I]] = null
  private final var populationInitializer: PopulationInitializer[I] = null
  private var populationSize: Int = 0
  private var logger: EvolutionLogger[I] = null
  private var generations: Int = 0

  def this(fitnessCalculator: FitnessCalculator[I], selectionStrategy: SelectionStrategy[I], crossover: Crossover[I], mutators: List[_ <: Mutator[I]], populationInitializer: PopulationInitializer[I], generations: Int, populationSize: Int) {
    this()
    this.fitnessCalculator = fitnessCalculator
    this.selectionStrategy = selectionStrategy
    this.crossover = crossover
    this.mutators = mutators
    this.populationInitializer = populationInitializer
    this.generations = generations
    this.populationSize = populationSize
  }

  def getFitnessCalculator: FitnessCalculator[I] = {
     fitnessCalculator
  }

  def getSelectionStrategy: SelectionStrategy[I] = {
     selectionStrategy
  }

  def getCrossover: Crossover[I] = {
     crossover
  }

  def getMutators: List[Mutator[I]] = {
     mutators
  }

  def getPopulationInitializer: PopulationInitializer[I] = {
     populationInitializer
  }

  def isLoggingEnable: Boolean = {
     logger != null
  }

  def getLogger: EvolutionLogger[I] = {
     logger
  }

  def setLogger(logger: EvolutionLogger[I]) {
    this.logger = logger
  }

  def getPopulationSize: Int = {
     populationSize
  }

  def setPopulationSize(populationSize: Int) {
    this.populationSize = populationSize
  }

  def getGenerations: Int = {
     generations
  }

  def setGenerations(generations: Int) {
    this.generations = generations
  }
}